pyfr/solvers/euler/kernels/rsolvers/exact.mako

```diff
@@ -0,0 +1,211 @@
+<%namespace module='pyfr.backends.base.makoutil' name='pyfr'/>
+<%include file='pyfr.solvers.euler.kernels.flux'/>
+
+<% gmrtg = (c['gamma'] - 1)/(2*c['gamma']) %>
+<% gprtg = (c['gamma'] + 1)/(2*c['gamma']) %>
+<% tgrgm = (2*c['gamma'])/(c['gamma'] - 1) %>
+<% trgm = 2/(c['gamma'] - 1) %>
+<% trgp = 2/(c['gamma'] + 1) %>
+<% gmrgp = (c['gamma'] - 1)/(c['gamma'] + 1) %>
+<% hgm = 0.5*(c['gamma'] - 1) %>
+<% tmgrgm = (2 - c['gamma'])/(c['gamma'] - 1) %>
+<% gamma = c['gamma'] %>
+
+// Initial guess for pressure
+<%pyfr:macro name='init_p' params='rl, vl, pl, cl, rr, vr, pr, cr, p0'>
+    fpdtype_t bpv = 0.5*(pl + pr) + 0.125*(vl[0] - vr[0])*(rl + rr)*(cl + cr);
+    bpv = max(0, bpv);
+    fpdtype_t pmin = min(pl, pr);
+    fpdtype_t pmax = max(pl, pr);
+
+    if ((pmax <= 2*pmin) && (pmin <= bpv) && (bpv <= pmax))
+    {
+        p0 = bpv;
+    }
+    else if (bpv < pmin)
+    {
+        // Two-rarefaction case
+        fpdtype_t pre = pow(pl / pr, ${gmrtg});
+        fpdtype_t um  = (pre*vl[0]*cr + vr[0]*cl + ${trgm}*(pre - 1)*cl*cr) /
+                        (pre*cr + cl);
+        fpdtype_t ptl = 1 - ${hgm}*(um - vl[0]) / cl;
+        fpdtype_t ptr = 1 + ${hgm}*(um - vr[0]) / cr;
+
+        p0 = 0.5*(pl*pow(ptl, ${tgrgm}) + pr*pow(ptr, ${tgrgm}));
+    }
+    else
+    {
+        // Two-shock case
+        fpdtype_t gl = sqrt(${trgp} / (rl*(${gmrgp}*pl + bpv)));
+        fpdtype_t gr = sqrt(${trgp} / (rr*(${gmrgp}*pr + bpv)));
+        p0 = (gl*pl + gr*pr - vr[0] + vl[0]) / (gl + gr);
+    }
+</%pyfr:macro>
+
+// Star Flux, assuming covolume = 0. See Toro 2009 Eq.(4.86 - 4.87)
+<%pyfr:macro name='star_flux' params='p, ps, rs, cs, f, fd'>
+    if (p <= ps)
+    {
+       fpdtype_t pr = p / ps;
+       f  = ${trgm}*cs*(pow(pr, ${gmrtg}) - 1);
+       fd = pow(pr, ${-gprtg}) / (rs*cs);
+    }
+    else
+    {
+       fpdtype_t bs = ${gmrgp}*ps;
+       fpdtype_t pbs_inv = 1 / (p + bs);
+       fpdtype_t sapb = sqrt(${trgp} * pbs_inv / rs);
+       f  = (p - ps)*sapb;
+       fd = sapb - 0.5*sapb*(p - ps) * pbs_inv;
+    }
+</%pyfr:macro>
+
+// Exact solve solution decision tree
+<% switch = 0 %>
```

Comment:
?

Reply:
See toro. This is the point on x/t to sample, this is what people normally do, but I don't see a reason wy we would ever want it to be anything other than 0.

Reply:
I think you would only use 0 for interface Riemann solvers and other values if you want to compute exact solutions, right?

Reply:
Yep, I've removed this setting

Reply:
It should allow some of the conditions involving `${switch}` to become more natural.